common/eth2/src/types.rs

```diff
@@ -1171,37 +1171,60 @@ impl FromStr for Accept {
     type Err = String;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let mut mimes = parse_accept(s)?;
+        let media_type_list = MediaTypeList::new(s);
 
         // [q-factor weighting]: https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.2
         // find the highest q-factor supported accept type
-        mimes.sort_by_key(|m| {
-            Reverse(m.get_param("q").map_or(1000_u16, |n| {
-                (n.as_ref().parse::<f32>().unwrap_or(0_f32) * 1000_f32) as u16
-            }))
+        let mut highest_q = 0_u16;
+        let mut accept_type = None;
+
+        const _APPLICATION: &str = APPLICATION.as_str();
+        const _OCTET_STREAM: &str = OCTET_STREAM.as_str();
+        const _JSON: &str = JSON.as_str();
+        const __STAR: &str = _STAR.as_str();
+        const _Q: &str = Q.as_str();
```

Comment:
Rather than using these `_` names, can we instead avoid importing the bare names and refer to them by path, like this:

```rust
use mediatype::names;

const APPLICATION: &str = names::APPLICATION.as_str();
```

Reply:
Hi @michaelsproul, I have made these changes.